### ISSUE TITLE
Lokinet API

### ISSUE THREAD

**Issue by sh4m2hwz** [oxen-io/lokinet#1771]
where can I see the api and are there examples of the client and server model, that is, full integration of the lokinet library and the shell that communicates with the service (that is, you can call the server that is currently providing information to the client)?
is there a separate lokinet library, or what needs to be cut from the lokinet repository approximately to get a full-fledged daemon-independent library?

**Comment by majestrate** [comment]
lokinet does not currently have an official publicly facing APIs in the form of a library, one may appear in the future. the general idea is to run it with a tun interface and map .loki onto the range owned by the tun interface using dns. i do not personally think the existence of such an api has enough legitimate use cases to warrant it.

**Comment by sh4m2hwz** [comment]
well, then there is an option to take a binary and run it on behalf of the current process with a virtual file system in ram

**Comment by majestrate** [comment]
sounds like a job for systemd or your init scripts.

**Comment by sh4m2hwz** [comment]
in order not to create files, but to integrate it as a built-in service into the application, for example, a browser that has a built-in loki, so that it is out of the box

**Comment by sh4m2hwz** [comment]
Yes it is very crazy))

**Comment by majestrate** [comment]
currently, the lokinet process needs `cap_net_admin` and `cap_bind_service` to function.

**Comment by sh4m2hwz** [comment]
but perhaps this is an interesting experience :)

**Comment by majestrate** [comment]
having a totally ephemeral mode where it doesn't touch the filesystem after setup is possible but it would suck as it would have to start from zero and rebootstrap every time it starts up.

**Comment by sh4m2hwz** [comment]
yes, but I think you can make friends with the lokinet on ramfs together with the browser and make a launcher that will launch lokinet and the browser

**Comment by sh4m2hwz** [comment]
well, you can make a separate project that compiles your project, then converts to PIE shellcode adds a loader if desired, and shells over lokinet, such as websockets, sockets, smtp, etc., and arrange it in lib

**Comment by sh4m2hwz** [comment]
this is a faster way than digging sorts, the same petoshellcode or execute in memory

**Comment by majestrate** [comment]
that would require so much bullshit glue code to make work, it comes off as incredibly tacky style.

**Comment by sh4m2hwz** [comment]
perception of everyone's own -)
